lms/djangoapps/grades/api/views.py

```diff
@@ -171,6 +186,18 @@ def get(self, request, course_id):
         Return:
             A JSON serialized representation of the requesting user's current grade status.
         """
+        # See if the request has an explicit sattr(request, 'allowed_organizations'))
+        # which limits which OAuth2 clients can see the courses
+        # based on the association with a RestrictedApplication
+        if hasattr(request, 'auth') and hasattr(request, 'oauth_scopes_filters'):
+            course_key = CourseKey.from_string(course_id)
+            if 'content_org' in request.oauth_scopes_filters.keys():
+                if course_key.org not in request.oauth_scopes_filters['content_org']:
+                    return self.make_error_response(
+                        status_code=status.HTTP_403_FORBIDDEN,
+                        developer_message='The OAuth2 RestrictedApplication is not associated with org.',
+                        error_code='course_org_not_associated_with_calling_application'
+                    )
```

Comment:
This would be good in some kind of scope enforcing decorator possibly.

Reply:
Possibly a DRF [filter](http://www.django-rest-framework.org/api-guide/filtering/).